### PR TITLE
STM32 HW-test die/digital tempearture Sensor in CI

### PR DESCRIPTION
**Pull request by djiatsaf-st**
- Add overlay files for some specifics boards to be able to run on HW with die_temp_polling sample

- Update sample.yaml to integrate new boards in CI

### CODE REVIEW

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
@avisconti @erwango please take a look

File: samples/sensor/die_temp_polling/sample.yaml [hunk 1]
```diff
@@ -7,8 +7,26 @@ tests:
...
+    - nucleo_g474re
+    - nucleo_h7a3zi_q
+    - nucleo_h743zi
+    - nucleo_l073rz
+    - nucleo_l152re
+    - nucleo_wba55cg
+    - nucleo_wb55rg
+    - nucleo_wl55jc
+    - stm32f3_disco
+    - stm32h573i_dk
```
[the remark this answers is not in the document]
Comment (djiatsaf-st) [reply samples/sensor/die_temp_polling/sample.yaml]: Thanks, good point, I'll use platform_allow instead.

File: samples/sensor/die_temp_polling/boards/b_u585i_iot02a.overlay [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+&die_temp {
+	status = "okay";
+};
+
+&adc1 {
+	pinctrl-0 = <&adc1_in5_pa0>;
+	pinctrl-names = "default";
+	st,adc-prescaler = <4>;
+	status = "okay";
+};
```
[the remark this answers is not in the document]
Comment (djiatsaf-st) [reply samples/sensor/die_temp_polling/boards/b_u585i_iot02a.overlay:22]: Not really, I'll put it in the devicetree

File: boards/st/nucleo_h743zi/nucleo_h743zi.dts [hunk 3]
```diff
@@ -160,7 +160,7 @@ zephyr_udc0: &usbotg_fs {
 };
 
 &adc3 {
-	pinctrl-0 = <&adc3_inp5_pf3>;
+	pinctrl-0 = <&adc3_inp5_pf3 &adc3_inp0_pc2_c>;
```
Comment (gautierg-st) [inline boards/st/nucleo_h743zi/nucleo_h743zi.dts:163]: PC2_C is a special pin that may require a call to the LL_SYSCFG_OpenAnalogSwitch function if PC2 come to be used for something else (to separate the 2 pads). Is there a need to add this specific pin?

File: boards/st/nucleo_wba55cg/nucleo_wba55cg.dts [hunk 4]
```diff
@@ -62,6 +62,18 @@
...
+		compatible = "st,stm32-temp-cal";
+		ts-cal1-addr = <0x0BF90710>;
+		ts-cal2-addr = <0x0BF90742>;
+		ts-cal1-temp = <30>;
+		ts-cal2-temp = <130>;
+		ts-cal-vrefanalog = <3000>;
+		io-channels = <&adc4 13>;
+		status = "okay";
+	};
+
```
Comment (gautierg-st) [inline boards/st/nucleo_wba55cg/nucleo_wba55cg.dts:76]: This should be in the .dtsi file (except that the status should be `"disabled"` in it).
You should only have `status = "okay";` here.

File: boards/st/stm32h573i_dk/stm32h573i_dk.dts [hunk 5]
```diff
@@ -53,11 +53,22 @@
...
+		digi_die_temp: digi_dietemp@40008C00 {
+			compatible = "st,stm32-digi-temp";
+			reg = <0x40008C00 0x400>;
+			interrupts = <113 0>;
+			interrupt-names = "digi";
+			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00000008>;
+			status = "okay";
+		};
+	};
+
```
Comment (gautierg-st) [inline boards/st/stm32h573i_dk/stm32h573i_dk.dts:66]: Same remark as for the WBA. Should be in dtsi.

**Review by gautierg-st — CHANGES_REQUESTED** [review]

File: dts/arm/st/wba/stm32wba55.dtsi [hunk 6]
```diff
@@ -9,4 +9,16 @@
...
+
+		compatible = "st,stm32-temp-cal";
+		ts-cal1-addr = <0x0BF90710>;
+		ts-cal2-addr = <0x0BF90742>;
+		ts-cal1-temp = <30>;
+		ts-cal2-temp = <130>;
+		ts-cal-vrefanalog = <3000>;
+		io-channels = <&adc4 13>;
+		status = "disabled";
+	};
```
Comment (gautierg-st) [inline dts/arm/st/wba/stm32wba55.dtsi:23]: This should also be added in stm32wba.dtsi.
Nit: remove the blank line.

File: boards/st/disco_l475_iot1/disco_l475_iot1.dts [hunk 7]
```diff
@@ -295,7 +296,7 @@ zephyr_udc0: &usbotg_fs {
 
 &adc1 {
 	pinctrl-0 = <&adc1_in3_pc2 &adc1_in4_pc3
-		     &adc1_in13_pc4 &adc1_in14_pc5>;
+		     &adc1_in13_pc4 &adc1_in14_pc5 >;
```
Comment (gautierg-st) [inline boards/st/disco_l475_iot1/disco_l475_iot1.dts:299]: Nit: remove the space before >

**Review by gautierg-st — CHANGES_REQUESTED** [review]
Still a few comments :)

File: boards/st/stm32h573i_dk/stm32h573i_dk.dts [hunk 8]
```diff
@@ -57,7 +57,7 @@
 		led0 = &blue_led_0;
 		sw0 = &user_button;
 		watchdog0 = &iwdg;
-		die-temp0 = &die_temp;
+		die-temp0 = &digi_die_temp;
```
[the remark this answers is not in the document]
Comment (djiatsaf-st) [reply boards/st/stm32h573i_dk/stm32h573i_dk.dts:60]: Good point , i will use this configuration instead.  

```
die-temp0 = &digi_die_temp; 
die-temp1 = &die_temp;
```

The sample.yaml have a filter on the aliase "die-temp0" , and the purpose of this test is about digital-temp first .

will this good for you ?

[the remark this answers is not in the document]
Comment (gautierg-st) [reply boards/st/stm32h573i_dk/stm32h573i_dk.dts:60]: It doesn't change anything, both sensors will be tested by the sample. The filter looks for the first (index 0), but the code will test all that follow.

File: dts/arm/st/h5/stm32h5.dtsi [hunk 9]
```diff
@@ -595,7 +595,7 @@
 			compatible = "st,stm32-digi-temp";
 			reg = <0x40008c00 0x400>;
 			interrupts = <113 0>;
-			interrupt-names = "digi";
+			interrupt-names = "digi_temp";
```
Comment (MaureenHelm) [inline dts/arm/st/h5/stm32h5.dtsi:598]: This change belongs in commit `dts: arm: st: add digi-temp node in dtsi file`

File: samples/sensor/die_temp_polling/boards/nucleo_u575zi_q.overlay [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+
+/ {
+	aliases {
+		die-temp0 = &die_temp;
+	};
+};
+
+&die_temp {
+	status = "okay";
+};
```
Comment (MaureenHelm) [inline samples/sensor/die_temp_polling/boards/nucleo_u575zi_q.overlay:16]: These changes can be moved to the board's dts, don't need an overlay.

File: samples/sensor/die_temp_polling/boards/stm32h750b_dk.overlay [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+};
+
+&adc3 {
+	pinctrl-0 = <&adc3_inn3_pf8>;
+	pinctrl-names = "default";
+
+	st,adc-clock-source = <SYNC>;
+	st,adc-prescaler = <4>;
+	status = "okay";
+};
```
Comment (MaureenHelm) [inline samples/sensor/die_temp_polling/boards/stm32h750b_dk.overlay:24]: These changes can be moved to the board's dts, don't need an overlay.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Review by gautierg-st — CHANGES_REQUESTED** [review]

**Review by gautierg-st — DISMISSED** [review]

**Review by gautierg-st — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**